python/pyarrow/_parquet.pyx

```diff
@@ -880,6 +880,21 @@ cdef encoding_name_from_enum(ParquetEncoding encoding_):
     }.get(encoding_, 'UNKNOWN')
 
 
+cdef encoding_enum_from_name(str encoding_name):
+    enc = {
+        'PLAIN': ParquetEncoding_PLAIN,
+        'BIT_PACKED': ParquetEncoding_BIT_PACKED,
+        'RLE': ParquetEncoding_RLE,
+        'BYTE_STREAM_SPLIT': ParquetEncoding_BYTE_STREAM_SPLIT,
+        'DELTA_BINARY_PACKED': ParquetEncoding_DELTA_BINARY_PACKED,
+        'DELTA_BYTE_ARRAY': ParquetEncoding_DELTA_BYTE_ARRAY,
+    }.get(encoding_name, None)
+    if enc is None:
+        raise ValueError(f"Unsupported column encoding: {encoding_name!r}")
```

Comment:
I think we should maybe still check for dictionary encoding specifically, and raise a slightly different error message.

Because right now, if people pass a dictionary encoding here, you get "Unsupported column encoding", while it's not actually unsupported. Only unsupported through this keyword, since it's already used by default. So the current error message might be quite confusing for users when they do that.

Reply:
What if I do something like this:

```
cdef encoding_enum_from_name(str encoding_name):
    enc = {
        'PLAIN': ParquetEncoding_PLAIN,
        'BIT_PACKED': ParquetEncoding_BIT_PACKED,
        'RLE': ParquetEncoding_RLE,
        'BYTE_STREAM_SPLIT': ParquetEncoding_BYTE_STREAM_SPLIT,
        'DELTA_BINARY_PACKED': ParquetEncoding_DELTA_BINARY_PACKED,
        'DELTA_BYTE_ARRAY': ParquetEncoding_DELTA_BYTE_ARRAY,
        'RLE_DICTIONARY': 'dict',
        'PLAIN_DICTIONARY': 'dict',
    }.get(encoding_name, None)
    if enc is None:
        raise ValueError(f"Unsupported column encoding: {encoding_name!r}")
    elif enc == 'dict':
        raise ValueError(f"{encoding_name!r} is already used by default.")
    else:
        return enc
```

Reply:
Yes, something like that sounds good

Reply:
BIT_PACKED I believe is [deprecated](https://github.com/apache/parquet-format/blob/master/Encodings.md#bit-packed-deprecated-bit_packed--4)

Reply:
IIRC the "BIT_PACKED" encoding is also only supported for repetition/definition levels, while the encoding specified here is for the actual values. 
So we could also not include it in the list below and thus give the "Unsupported column encoding" message from the snippet above. By leaving it in here, we bubble up the error message that the C++ parquet code will raise (from https://github.com/apache/arrow/blob/2baed02d259fe3534d20de14af6f7df48611b858/cpp/src/parquet/encoding.cc#L2643-L2705)